### ISSUE TITLE
detect.py

### ISSUE THREAD

**Issue by alicera** [ultralytics/yolov5#192]
why you run twice inference?
 

>    re = model(img.half() if half else img) if device.type != 'cpu' else None  # run once      
  pred = model(img, augment=opt.augment)[0]

why you do the GN
 

> gn = torch.tensor(im0.shape)[[1, 0, 1, 0]]  #  normalization gain whwh
 xywh = (xyxy2xywh(torch.tensor(xyxy).view(1, 4)) / gn).view(-1).tolist()  # normalized xywh

**Comment by glenn-jocher** [comment]
@alicera initial run produces more accurate times later.

gn is there for a purpose, remove it to see what happens.

**Comment by dewball345** [comment]
@glenn-jocher what is gn there for? I tried converting your python implementation to javascript but I didn't use gn; my bounding boxes were slightly dilated(or just offset)

**Comment by glenn-jocher** [comment]
@dewball345 gn is gain normalization as the comment states.

**Comment by dewball345** [comment]
@glenn-jocher well, what is gain normalization? I could see the comment but I don't understand what it is. I searched on google and am just getting audio or something. It looks to me that you are applying a transformation to the detection?
